src/main/java/io/vertx/core/AsyncResult.java

```diff
@@ -59,4 +61,44 @@
    */
   boolean failed();
 
+  /**
+   * Apply a {@code mapper} function on this async result.<p>
+   *
+   * The {@code mapper} is called with the completed value and this mapper returns a value. This value will complete the result returned by this method call.<p>
+   *
+   * If the {@code mapper} throws an exception, the returned future will be failed with this exception.<p>
+   *
+   * When this async result is failed, the failure will be propagated to the returned future and the {@code mapper}
+   * will not be called.
+   *
+   * @param mapper the mapper function
+   * @return the mapped async result
+   */
+  <U> AsyncResult<U> map(Function<T, U> mapper);
```

Comment:
Good to expose those methods already present in Future in the AR interface (since it's what a vast majority of our APIs use).